server-common/src/main/java/org/apache/kafka/server/common/MetadataVersion.java

```diff
@@ -228,7 +231,7 @@ public enum MetadataVersion {
      * <strong>Think carefully before you update this value. ONCE A METADATA VERSION IS PRODUCTION,
      * IT CANNOT BE CHANGED.</strong>
      */
-    public static final MetadataVersion LATEST_PRODUCTION = IBP_3_7_IV4;
+    public static final MetadataVersion LATEST_PRODUCTION = IBP_3_8_IV1;
```

Comment:
Not yet. We are still finalizing the last PR. https://github.com/apache/kafka/pull/15702
@clolov I don't want to block your feature, we should mark KIP-966 enabled for another metadata version.
@junrao What is your suggestion here? Using a future metadata version like IBP_4_0_IV0 for KIP-966? Also, should I make a change separately or we can do it in this PR?

Reply:
Not yet. We are still finalizing the last PR. https://github.com/apache/kafka/pull/15702
@clolov I don't want to block your feature, we should let KIP-966 use another metadata version and mark the IBP_3_8_IV0 as reserved.
@junrao What is your suggestion here? Using a future metadata version like IBP_4_0_IV0 for KIP-966 at the moment?

Reply:
Thanks, @CalvinConfluent .

@clolov : In that case, we will need to (1) mark IBP_3_8_IV0 as unused, (2) create a new MV IBP_4_0_IV0, (3) replace existing references of IBP_3_8_IV0  related to Elr like the following to IBP_4_0_IV0.

```
    public boolean isElrSupported() {
        return this.isAtLeast(IBP_3_8_IV0);
    }
```

Reply:
The 4.0 IV0 is taken, I think we should use IV1?
```
// Introduce version 1 of the GroupVersion feature (KIP-848).
    IBP_4_0_IVO(21, "4.0", "IV0", false);
```

Reply:
Okay, but can the people working on ELR ensure they reserve the new IBP as part of their work? I want to constrain this PR to just allocate IBP_3_8_IV1 for 1005 to be completed

Reply:
Hey folks, the point of unstable metadata versions is that we should be able to move stuff around and not have to keep creating new ones. 😅

I was hoping that KIP would be finalized so we could move forward with this protocol.

Reply:
@jolshan : Just to clarify. Are you saying that ELR can just use MV 4.0 IV0, instead of creating a new one?

Reply:
If ELR is being pushed to 4.0 it may be worth making 4.0-IV1 since both the group version and the ELR are not stable.

But yes, I do mean that if we plan to release a new MV in 3.8, we should just use 3.8-IV0.

Reply:
@jolshan: What's the benefit of reusing 3.8-IV0 for ListOffset instead of creating a new 3.8-IV1? To me, creating 3.8-IV1 makes it a bit easier to understand the change history. It leaves an IV unused, but that doesn't seem a big concern.

Reply:
I'm not sure I understand the benefit of leaving it unused. I would suspect if there are many ongoing features, we will have a lot of unused versions and that is confusing it its own way. I was hoping the details of KIP-1014 would be finalized by now so that there was clear guidance for this sort of state.

Reply:
ELR was originally associated with IBP_3_7_IV3. When it's moved IBP_3_8_IV0, we left IBP_3_7_IV3 unused since IBP_3_7_IV4 has already been implemented. We didn't reuse IBP_3_7_IV3. Here, there is no 3.8 IV after IBP_3_8_IV0. So, we could reuse  IBP_3_8_IV0 for ListOffset. However, it seems that a simpler and more consistent policy is to never reuse an IV?

Reply:
Heh, this comment thread is becoming the KIP-1014 discussion. I suppose while we have not completed that discussion we can just pick an approach for now. I don't feel super strongly about this for the one more MV while we don't have the conclusive answer.

Reply:
There is really no reason to leave unstable MVs unused. I think the issue with IBP_3_7_IV3 was a misunderstanding. The whole point of an unstable MV is you can do whatever you want.